lib/portage/sync/modules/git/git.py

```diff
@@ -341,3 +346,29 @@ def retrieve_head(self, **kwargs):
         except subprocess.CalledProcessError:
             ret = (1, False)
         return ret
+
+    def add_safe_directory(self, directory):
+        # https://github.com/gentoo/portage/pull/818
+        # Workaround for bug #838271 and bug #838223
+        # We unset first to avoid duplicate entries and also to handle
+        # relocations.
+        cmds = [
+            [self.bin_command, "config", "--unset-all", "safe.directory"],
+            [self.bin_command, "config", "--add", "safe.directory", directory],
```

Comment:
This doesn't work because git ignores the setting in .git/config within a repo. We would need to pass `--global` to add it to root's .gitconfig, or `--system` to add it to /etc/gitconfig.

Reply:
I'll work on a patch to check the status of `git config --system --get` and call `git config --system --add`.

Reply:
I've got no idea why it seemed to work for me. Maybe my permissions were wonky. Thanks.